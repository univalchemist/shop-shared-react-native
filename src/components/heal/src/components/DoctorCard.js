import React from 'react';
import { TouchableOpacity, StyleSheet, Image } from 'react-native';
import { Box, Text } from '@heal/src/wrappers/components';
import { useIntl, useTheme } from '@heal/src/wrappers/core/hooks';
import { chevronRight, iconTime, doctorBanner } from '@heal/images';
import theme from '@theme';

const styles = StyleSheet.create({
  chevron: { alignSelf: 'center' },
  image: {
    width: 188,
    height: 117,
  },
  card: {
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.05,
    shadowRadius: 6,
    shadowColor: theme.colors.black,
    backgroundColor: theme.colors.white,
  },
  selectedCard: {
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.05,
    shadowRadius: 6,
    shadowColor: theme.colors.black,
    backgroundColor: theme.colors.white,
    borderColor: theme.colors.green,
    borderWidth: 1,
  },
});

const DoctorCard = ({
  doctor,
  width,
  height,
  selected,
  icon,
  showIcon,
  onPress,
}) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableOpacity onPress={onPress}>
      <Box
        width={width}
        height={height}
        style={selected ? styles.selectedCard : styles.card}
      >
        <Image
          source={doctorBanner}
          style={styles.image}
          backgroundColor={theme.colors.gray[4]}
        />
        <Box flex={1} mx={16} my={16} flexDirection="row">
          <Box marginRight={16}>
            <Text numberOfLines={2} color="gray.0">
              {doctor.name}
            </Text>
            <Text numberOfLines={2} fontSize={14} lineHeight={20} mt={8}>
              {doctor.specialityCode
                ? intl.formatMessage({
                    id: `speciality.name.${doctor.specialityCode}`,
                  })
                : ''}
            </Text>
            <Box marginTop="auto" flexDirection="row">
              <Image source={iconTime} />
              <Text
                marginLeft={10}
                fontSize={12}
                lineHeight={20}
                color={theme.colors.gray[3]}
              >
                {`${doctor.estimatedConsultationTime} ${intl.formatMessage({
                  id: 'clinic.estimatedTime',
                })}`}
              </Text>
            </Box>
          </Box>
          {showIcon ? (
            <Image
              marginLeft="auto"
              source={icon ? icon : chevronRight}
              style={styles.chevron}
            />
          ) : null}
        </Box>
      </Box>
    </TouchableOpacity>
  );
};

export default DoctorCard;
