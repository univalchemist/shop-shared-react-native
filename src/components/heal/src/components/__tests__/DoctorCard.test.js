import React from 'react';
import { renderForTest } from '@testUtils';

import DoctorCard from '../DoctorCard';

const doctorData = {
  isOpen: true,
  isRemoteFull: false,
  isRemoteTicketEnabled: true,
  isAppointmentEnabled: true,
  position: 0,
  estimatedConsultationTime: '3:23 pm',
  ticketDepositAmount: 0.0,
  canGetRefundOnCancellation: false,
  cancellationGraceMinutes: 60,
  id: 322,
  name: null,
  specialityCode: null,
  gender: 'Male',
  email: null,
  imageUrl: null,
  introduction: null,
};

const DOCTOR_CARD_WIDTH = 190;
const DOCTOR_CARD_HEIGHT = 314;

describe('ClinicListingItem', () => {
  test('Render Clinic Listing Item', async () => {
    const component = renderForTest(
      <DoctorCard
        doctor={doctorData}
        width={DOCTOR_CARD_WIDTH}
        height={DOCTOR_CARD_HEIGHT}
      />,
      {
        initialState: {},
      },
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
