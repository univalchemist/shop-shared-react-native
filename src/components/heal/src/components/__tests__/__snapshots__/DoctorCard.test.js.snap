// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicListingItem Render Clinic Listing Item 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={false}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          height={314}
                          style={
                            Array [
                              Object {
                                "height": 314,
                                "width": 190,
                              },
                              Object {
                                "backgroundColor": "#ffffff",
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 0,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.05,
                                "shadowRadius": 6,
                              },
                            ]
                          }
                          width={190}
                        >
                          <Image
                            backgroundColor="#b8b8b8"
                            source={
                              Object {
                                "testUri": "../../../src/components/heal/images/doctor_panel.png",
                              }
                            }
                            style={
                              Object {
                                "height": 117,
                                "width": 188,
                              }
                            }
                          />
                          <View
                            flex={1}
                            flexDirection="row"
                            mx={16}
                            my={16}
                            style={
                              Array [
                                Object {
                                  "flexBasis": 0,
                                  "flexDirection": "row",
                                  "flexGrow": 1,
                                  "flexShrink": 1,
                                  "marginBottom": 16,
                                  "marginLeft": 16,
                                  "marginRight": 16,
                                  "marginTop": 16,
                                },
                              ]
                            }
                          >
                            <View
                              marginRight={16}
                              style={
                                Array [
                                  Object {
                                    "marginRight": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="gray.0"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                numberOfLines={2}
                                style={
                                  Array [
                                    Object {
                                      "color": "#212121",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              />
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={14}
                                lineHeight={20}
                                mt={8}
                                numberOfLines={2}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 14,
                                      "lineHeight": 20,
                                      "marginTop": 8,
                                    },
                                  ]
                                }
                              >
                                
                              </Text>
                              <View
                                flexDirection="row"
                                marginTop="auto"
                                style={
                                  Array [
                                    Object {
                                      "flexDirection": "row",
                                      "marginTop": "auto",
                                    },
                                  ]
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "testUri": "../../../src/components/heal/images/icon-time.png",
                                    }
                                  }
                                />
                                <Text
                                  color="#9a9a9a"
                                  fontFamily="default"
                                  fontSize={12}
                                  lineHeight={20}
                                  marginLeft={10}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#9a9a9a",
                                        "fontFamily": "Arial",
                                        "fontSize": 12,
                                        "lineHeight": 20,
                                        "marginLeft": 10,
                                      },
                                    ]
                                  }
                                >
                                  3:23 pm estimated consulation time
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
